Implemented Controller Hotkeys

AdrianCassar opened the pull request:
Added controller hotkeys
Added guide button support for XInput and winkey

The hotkey configurations can be found in **HID -> Display controller hotkeys**.
If the Xbox Gamebar overlay is enabled then use the **Back** button instead of the **Guide** button.

AdrianCassar committed:
Implemented Controller Hotkeys

Added controller hotkeys
Added guide button support for XInput and winkey

The hotkey configurations can be found in HID -> Display controller hotkeys
If the Xbox Gamebar overlay is enabled then use the Back button instead of the Guide button.

Gliniak commented on src/xenia/app/emulator_window.cc:
```diff
@@ -1071,10 +1094,248 @@ void EmulatorWindow::SetInitializingShaderStorage(bool initializing) {
+  DWORD value = 0;
+  DWORD dataSize = sizeof(value);
+
+  RegGetValue(HKEY_CURRENT_USER, reg_path, key, RRF_RT_DWORD, nullptr, &value,
```
I guess it would be nice to check if function failed and returned error code. I suspect that someone might not have it and it will cause confusion

Gliniak reviewed (approved):
+1, waiting for Chrispy review

AdrianCassar committed:
Minor bug fixes for controller hotkeys

- Fixed hotkey thread destruction
- Fixed XINPUT_STATE by padding 4 bytes
- Added hotkey vibration for user feedback
- Replaced MessageBoxA with ImGuiDialog::ShowMessageBox

AdrianCassar committed:
Minor Improvements

Margen67 commented on src/xenia/app/emulator_window.cc:
```diff
@@ -1071,26 +1095,308 @@ void EmulatorWindow::SetInitializingShaderStorage(bool initializing) {
+         EmulatorWindow::ButtonFunctions::ToggleControllerVibration,
+         "Right Shoulder + Guide = Toggle Controller Vibration", true)},
+
+    // CPU Time Scalar with no rubmle feedback
```
```suggestion
    // CPU Time Scalar with no rumble feedback
```

AdrianCassar committed:
Update src/xenia/app/emulator_window.cc

Fixed a typo.

Co-authored-by: Margen67 <Margen67@users.noreply.github.com>
